scanapp/views.py

```diff
@@ -151,3 +165,147 @@ def post(self, request):
                 }
             )
         )
+
+
+class URLFormView(FormView):
+    template_name = 'scanapp/urlspecialscan.html'
+    form_class = URLScanForm
+
+    def post(self, request, *args, **kwargs):
+        form = self.form_class(request.POST)
+
+        if form.is_valid():
+            # get the URL from the form
+            URL = request.POST['URL']
+
+            scan_type = 'URL'
+
+            allowed_exts = ('git')
+            url = urlparse(URL)
+
+            # Create an Instance of InsertIntoDB
+            insert_into_db = InsertIntoDB()
+
+            # call the create_scan_id function
+            scan_id = insert_into_db.create_scan_id(scan_type)
+
+            if url.path.rsplit('.', 1)[1] in allowed_exts:
+                print 'git repo'
+
+                dir_name = 'temp'
+
+                home_path = expanduser("~")
+
+                os.chdir(home_path)
+
+                if os.path.isdir(dir_name):
+                    shutil.rmtree(dir_name)
+
+                os.mkdir(dir_name)
+
+                repo = git.Repo.init(dir_name)
+                origin = repo.create_remote('origin', URL)
+                origin.fetch()
+                origin.pull(origin.refs[0].remote_head)
+
+                print "Done ! Remote repository cloned."
+
+                # Create an Instance of InsertIntoDB
+                insert_into_db = InsertIntoDB()
+
+                # call the create_scan_id function
+                scan_id = insert_into_db.create_scan_id(scan_type)
+
+                filename = home_path + '/temp/'
+
+                # print filename
+
+                path = filename
+
+                folder_name = None
+
+                apply_scan_async.delay(path, scan_id, scan_type, URL, folder_name)
+
+                # return the response as HttpResponse
+                return HttpResponseRedirect('/resultscan/' + str(scan_id))
+
+            else:
+
+                # different paths for both anonymous and registered users
+                if (str(request.user) == 'AnonymousUser'):
+                    path = 'media/AnonymousUser/URL/'
+
+                else:
+                    path = 'media/user/' + str(request.user) + '/URL/'
+
+                scan_code_async.delay(URL, scan_id, path)
+                # return the response as HttpResponse
+                return HttpResponseRedirect('/resultscan/' + str(scan_id))
+
+
+class UrlScanSpecialView(FormView):
+    template_name = 'scanapp/urlspecialscan.html'
```

Comment:
I had a discussion with @singh1114 where we planned to have a new view for the URL Scan and keep the URLSpecialScan intact for testing purposes which we would be removing after the main view is created.